### PR TITLE
[SPARK-38747][SQL][TESTS] Move the tests for `PARSE_SYNTAX_ERROR` from ErrorParserSuite  to QueryParsingErrorsSuite

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This PR aims to move tests for the error class PARSE_SYNTAX_ERROR from ErrorParserSuite to QueryParsingErrorsSuite.

### Why are the changes needed?
To improve code maintenance.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
By running the moved tests:
```
$ build/sbt "sql/testOnly *QueryParsingErrorsSuite*"
```

### CODE REVIEW

**Review by MaxGekk — COMMENTED** [review]
The `PARSE_SYNTAX_ERROR` error class is checked in other tests that you don't move. If you are not going to move them, explain in PR's description why do you think we shouldn't move them.

**Comment by panbingkun** [comment]
> The `PARSE_SYNTAX_ERROR` error class is checked in other tests that you don't move. If you are not going to move them, explain in PR's description why do you think we shouldn't move them.

I miss some 'syntax error' case, now i move them.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala [hunk 1]
```diff
@@ -472,4 +472,151 @@ class QueryParsingErrorsSuite extends QueryTest with SharedSparkSession {
...
+      sqlState = "42000",
+      message =
+        s"""
+           |Syntax error at or near 't': extra input 't'(line 2, pos 12)
+           |
+           |== SQL ==
+           |select *
+           |from r as q t
+           |------------^^^
+           |""".stripMargin)
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala:518]: 2 spaces:
```suggestion
        s"""
          |Syntax error at or near 't': extra input 't'(line 2, pos 12)
          |
          |== SQL ==
          |select *
          |from r as q t
          |------------^^^
          |""".stripMargin)
```

**Comment by MaxGekk** [comment]
Waiting for https://github.com/panbingkun/spark/actions/runs/2185240902

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**